SimFastTiming/FastTimingCommon/interface/MTDDigitizer.h

```diff
@@ -156,20 +227,33 @@ namespace mtd_digitizer {
     hitRefs.clear();
 
   }
+
+  template<class Traits>
+  void MTDDigitizer<Traits>::accumulate(const PMTDSimAccumulator& simAccumulator) {
+    loadSimHitAccumulator(simHitAccumulator_, simAccumulator, premixStage1MinCharge_, premixStage1MaxCharge_);
+  }
   
   template<class Traits>
   void MTDDigitizer<Traits>::initializeEvent(edm::Event const& e, edm::EventSetup const& c) {
     deviceSim_.getEvent(e);
-    electronicsSim_.getEvent(e);
+    if(not premixStage1_) {
+      electronicsSim_.getEvent(e);
+    }
   }
   
   template<class Traits>
   void MTDDigitizer<Traits>::finalizeEvent(edm::Event& e, edm::EventSetup const& c, 
 					   CLHEP::HepRandomEngine* hre) {
-    
-    auto digiCollection = std::make_unique<DigiCollection>();
-    electronicsSim_.run(simHitAccumulator_,*digiCollection, hre);
-    e.put(std::move(digiCollection),digiCollection_);
+    if(premixStage1_) {
+      auto simResult = std::make_unique<PMTDSimAccumulator>();
+      saveSimHitAccumulator(*simResult, simHitAccumulator_, premixStage1MinCharge_, premixStage1MaxCharge_); // FIXME
```

Comment:
what needs to be fixed?

Reply:
Nothing, the comment was a leftover from earlier hacking. Thanks for catching.